tools/genomespace/genomespace_importer.py

```diff
@@ -1,220 +1,265 @@
-# Dan Blankenberg
-
+import argparse
 import json
-import optparse
 import os
-import shutil
-import tempfile
+import sys
+import uuid
 
-from six.moves import http_cookiejar
-from six.moves.urllib.parse import parse_qs, unquote_plus, urlparse
-from six.moves.urllib.request import build_opener, HTTPCookieProcessor, Request, urlopen
+from genomespaceclient import GenomeSpaceClient
 
+import galaxy
 from galaxy.datatypes import sniff
 from galaxy.datatypes.registry import Registry
 
-GENOMESPACE_API_VERSION_STRING = "v1.0"
-GENOMESPACE_SERVER_URL_PROPERTIES = "https://dm.genomespace.org/config/%s/serverurl.properties" % ( GENOMESPACE_API_VERSION_STRING )
-DEFAULT_GENOMESPACE_TOOLNAME = 'Galaxy'
-FILENAME_VALID_CHARS = '.-()[]0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ '
-
-CHUNK_SIZE = 2**20  # 1mb
-
-DEFAULT_GALAXY_EXT = "data"
 
-# genomespace format identifier is the URL
-GENOMESPACE_FORMAT_IDENTIFIER_TO_GENOMESPACE_EXT = {}  # TODO: fix this so it is not a global variable
-# TODO: we should use a better way to set up this mapping
+# Mappings for known genomespace formats to galaxy formats
 GENOMESPACE_EXT_TO_GALAXY_EXT = {'rifles': 'rifles',
                                  'lifes': 'lifes',
                                  'cn': 'cn',
-                                 'GTF': 'gtf',
+                                 'gtf': 'gtf',
                                  'res': 'res',
                                  'xcn': 'xcn',
                                  'lowercasetxt': 'lowercasetxt',
                                  'bed': 'bed',
-                                 'CBS': 'cbs',
+                                 'cbs': 'cbs',
                                  'genomicatab': 'genomicatab',
                                  'gxp': 'gxp',
                                  'reversedtxt': 'reversedtxt',
                                  'nowhitespace': 'nowhitespace',
                                  'unknown': 'unknown',
                                  'txt': 'txt',
                                  'uppercasetxt': 'uppercasetxt',
-                                 'GISTIC': 'gistic',
-                                 'GFF': 'gff',
+                                 'gistic': 'gistic',
+                                 'gff': 'gff',
                                  'gmt': 'gmt',
-                                 'gct': 'gct'}
+                                 'gct': 'gct',
+                                 'fasta': 'fasta',
+                                 'fna': 'fasta',
+                                 'fa': 'fasta',
+                                 'faa': 'fasta',
+                                 'ffn': 'fasta',
+                                 'fastq': 'fastqsanger',
+                                 'fq': 'fastqsanger',
+                                 'fqz': 'fastqsanger'}
```

Comment:
If I understand this implementation correctly, all datatypes that are sniffed successfully will not be modified by this mapping, so I'd remove all `fasta/q*`, `gff`, `gtf`, `*txt` etc. types. In fact I'd probably eliminate this step, or, if you want to force things like `gtf`/`gff`, which are difficult to sniff and where you are certain that the genomespace extensions is correct, you'd have to change the order for this to occur before letting galaxy sniff the datatype.